### PR TITLE
WIP: Rewrite document on URL rewriting

### PR DESCRIPTION
**Pull request by wolfgangmm**
Work in progress, do not merge yet.

### CODE REVIEW

**Comment by duncdrum** [comment]
is this still WIP?

**Comment by adamretter** [comment]
@duncdrum Afraid I am not sure I can help with the content of this... URL Rewriting is not my area of expertise.

File: src/main/xar-resources/data/urlrewrite/urlrewrite.xml [hunk 1]
```diff
@@ -19,6 +19,34 @@
...
+      HTML files, data and other resources, organized into a hierarchical structure. Direct links to such deeply nested resources would result in very long URIs. You may not want to expose those structures to the users, but provide
+      short and human-readable URIs instead.</para>
+    <para>Let's contemplate the following scenario: the XML documents your app should display through a web page are contained in a sub-collection
+      called <code>data</code>. For example, the document you are currently reading resides in <code>data/urlrewrite.xml</code> within the documentation collection. The direct URL pointing to this document would thus be <code>/exist/apps/doc/data/urlrewrite.xml</code>. We do not want to show the user the raw XML though. Instead users should get a properly formatted HTML version of the text, accessible through a simple URL like <code>/exist/apps/doc/urlrewrite</code>.</para>
+    <para>To achieve this we need a mechanism to map or rewrite a given URL to an application specific endpoint. So in the simplest case, calling
```
Comment (lguariento) [inline src/main/xar-resources/data/urlrewrite/urlrewrite.xml]: application-specific

File: src/main/xar-resources/data/urlrewrite/urlrewrite.xml [hunk 2]
```diff
@@ -19,6 +19,34 @@
...
+    <programlisting xmlns:xlink="http://www.w3.org/1999/xlink" language="xquery" xlink:href="listings/listing-1.txt"/>
+    <para>This example controller returns a simple <tag>dispatch</tag> fragment only. This fragment will be passed back to the URL rewriting framework once the controller XQuery has been executed. The <tag>forward</tag> element instructs the framework to call the URL <literal>modules/transform.xql</literal> relative to the collection in which the controller resides. It adds a request parameter, named <literal>doc</literal>, which translates the requested resource into an actual document path to be transformed by <literal>modules/transform.xql</literal>.</para>
+    <para>Real world controllers are typically more complicated and may contain arbitrary XQuery code to distinguish between different scenarios. Basically, the URL rewriting framework allows you to turn simple requests into complex processing pipelines, involving any number of steps. For example, let us split above <tag>dispatch</tag> fragment into a pipeline involving two steps:</para>
```
Comment (lguariento) [inline src/main/xar-resources/data/urlrewrite/urlrewrite.xml]: "let us split the above"

File: src/main/xar-resources/data/urlrewrite/listings/listing-1.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+xquery version "3.0";
```
Comment (lguariento) [inline src/main/xar-resources/data/urlrewrite/listings/listing-1.txt]: Why not 3.1, while we're at it?

**Comment by duncdrum** [comment]
see #333 
see eXist-db/exist#2657

**Comment by adamretter** [comment]
@wolfgangmm What's the status of this?

**Comment by adamretter** [comment]
So @wolfgang reports this is still a WIP as of today.

**Comment by duncdrum** [comment]
see #105 @wolfgangmm please rebase

**Comment by joewiz** [comment]
Superseded by #872.

**Pull request closed without merging**